esp-println/src/logger.rs

```diff
@@ -62,8 +62,23 @@ impl log::Log for EspLogger {
         #[cfg(not(feature = "colors"))]
         let reset = "";
 
+        #[cfg(feature = "timestamp")]
+        println!(
+            "{}{} ({}) - {}{}",
+            color,
+            record.level(),
+            unsafe { _esp_println_timestamp() },
+            record.args(),
+            reset
+        );
+        #[cfg(not(feature = "timestamp"))]
         println!("{}{} - {}{}", color, record.level(), record.args(), reset);
     }
 
     fn flush(&self) {}
 }
+
+#[cfg(feature = "timestamp")]
+extern "Rust" {
+    fn _esp_println_timestamp() -> u64;
```

Comment:
We need to document this, and how to use it (with and without esp-hal) in the crate docs.